Fix: prevent crash when search query contains spaces (URL-encode query string)

agaragou opened the pull request:
This patch fixes a crash occurring when a search query contains spaces or special characters (e.g. “leigh bardugo”).
The issue was caused by unescaped characters in the query string passed to Api.makeHttpRequest.

Added a proper urlencode() function in src/scraper.lua and replaced simple string.gsub(query, " ", "+") with safe URL encoding.

Now multi-word searches and queries with symbols work correctly and no longer crash KOReader.

Update scraper.lua

agaragou committed:
Update scraper.lua

Fix for url encode (now we can search with more than one word without crashing koreader)

LaiKash commented:
Hey @sozanov! I didn't find this issue. Any other specific query that crashes your koreader?

agaragou commented:
@LaiKash 
honestly, i don't even know what to reply..
just to test again i just reverted to the original src/scraper.lua, entered "lieght bardugo" (with a space), and koreader crashed (kindle paperwhite 2024)

to describe what went wrong, i looked at the search filters: languages english+russian, format epub.  

I changed the filter values to English only.. and everything worked. switched back to English+another language, including Russian, and it worked again. for some reason..

maybe I hit some bug, but i can't reproduce it anymore.  
but when i modified the file without changing the filters, everything worked correctly.  
looks like QA is needed here :))

shakogegia commented:
Can confirm that it crashes koreader when spaces included,
Also it doesn't hurt to include `urlencode` anyway

device: Kindle Oasis 10th gen
